integration-tests/src/test/java/com/myhome/controllers/PaymentControllerIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+package com.myhome.controllers;
+
+import com.myhome.MyHomeServiceApplication;
+import com.myhome.model.SchedulePaymentRequest;
+import com.myhome.model.SchedulePaymentResponse;
+import com.myhome.repositories.PaymentRepository;
+import helpers.TestUtils;
+import java.math.BigDecimal;
+import java.util.UUID;
+import org.assertj.core.util.BigDecimalComparator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(
+    classes = MyHomeServiceApplication.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+public class PaymentControllerIntegrationTest {
+
+  private static final String TEST_MEMBER_ID = "default-member-id-for-testing";
+  private static final String TEST_PAYMENT_TYPE = "test";
+  private static final String TEST_PAYMENT_DESCRIPTION = "test description";
+  private static final UUID TEST_PAYMENT_ID =
+      UUID.fromString("5d09efc3-c08d-4e52-9367-3157850ad9e7");
+
+  @Autowired
+  private TestRestTemplate testRestTemplate;
+
+  @Autowired
+  private Environment env;
+
+  @Autowired
+  private PaymentRepository paymentRepository;
+
+  private String userId;
+  private String loginToken;
+
+  @BeforeEach
+  public void setUp() {
+    ResponseEntity<Void> loginResponse = TestUtils.Login.performLogin(testRestTemplate, env);
```

Comment:
We do not need to call login endpoint. We can create JWT token manually as non expiring one.

Reply:
I understand. But by calling the login endpoint, we can guarantee that a regular flow will be workable for users interacting with the system outside of the controlled test environment. If we generate a JWT token directly, we either:

- potentially bypass any info that gets encoded into it by the real login and therefore deviate from testing functionality that a real user might see, or
- duplicate a good portion of the login logic and make sure all future changes need to be kept up-to-date, or
- maybe even both

If this was a unit test, I'd completely agree with you that it shouldn't be relying on external behavior. As this is instead an integration test, I went with simulating real calls for the full test.

Reply:
Okay I can agree to that 👍